android/capacitor/src/main/java/com/getcapacitor/CapacitorWebView.java

```diff
@@ -54,19 +56,37 @@ public boolean dispatchKeyEvent(KeyEvent event) {
         return super.dispatchKeyEvent(event);
     }
 
-    private void edgeToEdgeHandler() {
-        ViewCompat.setOnApplyWindowInsetsListener(this, (v, windowInsets) -> {
-            Insets insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemBars());
+    public void edgeToEdgeHandler(Bridge bridge) {
+        String configEdgeToEdge = bridge.getConfig().adjustMarginsForEdgeToEdge();
 
-            MarginLayoutParams mlp = (MarginLayoutParams) v.getLayoutParams();
-            mlp.leftMargin = insets.left;
-            mlp.bottomMargin = insets.bottom;
-            mlp.rightMargin = insets.right;
-            mlp.topMargin = insets.top;
-            v.setLayoutParams(mlp);
+        boolean autoMargins = false;
+        boolean forceMargins = Objects.equals(configEdgeToEdge, "force");
+        boolean disableMargins = Objects.equals(configEdgeToEdge, "disable");
 
-            // Don't pass window insets to children
-            return WindowInsetsCompat.CONSUMED;
-        });
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            boolean foundOptOut = false;
+            boolean optOutValue = false;
```

Comment:
Those can be initialized directly without first setting them to false

Reply:
Yeah, moved those inside the if without thinking.